src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1037,24 +1074,47 @@ namespace cryptonote
       if(tvc[i].m_added_to_pool)
         MDEBUG("tx added: " << results[i].hash);
     }
+
+    // relay those that are valid
+    NOTIFY_NEW_TRANSACTIONS::request r;
+    it = m_defered_txs.begin();
+    for (size_t i = 0; i < m_defered_txs.size(); i++, ++it) {
+      if (!results[i].res || it->do_not_relay || it->keeped_by_block || already_have[i])
```

Comment:
Why not use `tvc[i].m_verifivation_failed` and `tvc[i].m_should_be_relayed` for determining whether to relay or not, as was done in `handle_notify_new_transactions`?